src/app/components/translate-attributes/translate-attributes.component.ts

```diff
@@ -0,0 +1,341 @@
+          if (attr.id === attribute.id) {
+            return;
+          }
+          if (attr.key === attribute.key) {
+            throw new Error(
+              '2 or more keys are the same. Please make sure all keys are unique.',
+            );
+          }
+        });
+      });
+      console.log('Update & Create checks - complete');
+
+      console.log('promises', this.multipleActionsPromises);
+      return {
+        successful: true,
+        message: 'Successful',
+      };
+    } catch (err) {
+      console.log('ERR', err);
+      return {
+        successful: false,
+        message: err.message,
+      };
+    }
+  }
+
+  multipleActions(): void {
+    this.multipleActionsLocalAttributes.clear();
+    this.multipleActionsRemoteAttributes.clear();
+    console.log('multipleActionsFunc');
+
+    this.attributeTranslationService
+      .getAttributes(this.resource.id)
+      .then(async (res) => {
+        try {
+          this.errorMessage = '';
+          this.multipleActionsError = false;
+          this.multipleActionsResults = [];
+          this.saving = true;
+          this.multipleActionsPromises = [];
+
+          const checkDifferences = await this.checkRemoteResourceForDifferences(
+            res,
+          );
+          if (!checkDifferences.successful) {
+            throw new Error(checkDifferences.message);
+          }
+
+          // Sort promises
+          await this.mulitipleActionSortPromises();
+
+          // Need to await for this
+          const sendPromises = new Promise(async (resolve, reject) => {
+            console.log('Start Promises');
+            for (const promise of this.multipleActionsPromises) {
+              console.log('promise.data', promise.data);
+              if (promise.type === 'delete') {
+                await this.mulitipleActionDelete(promise);
+              } else if (promise.type === 'update') {
+                await this.mulitipleActionUpdate(promise);
+              } else if (promise.type === 'create') {
+                await this.mulitipleActionCreate(promise);
+              }
+            }
+            console.log('Had Error?', this.multipleActionsError);
+
+            if (this.multipleActionsError) {
+              reject('Error');
+            } else {
+              resolve('Completed');
+            }
+          });
+
+          await Promise.all([sendPromises]);
+          this.saving = false;
+          setTimeout(() => this.closeEditModal(), 2000);
+        } catch (err) {
+          this.handleError(err.message);
+        }
+      });
+  }
+
+  mulitipleActionSortPromises(): void {
+    // Sort promises
+    // 1. delete items first ()
+    // 2. update items second
+    // 3. create items last
+    this.multipleActionsPromises.sort((a, b) => {
+      if (b.type === 'delete') {
+        return 1;
+      }
+      if (a.type === 'delete') {
+        return -1;
+      }
+      if (a.type === 'create' && b.type === 'create') {
+        return -1;
+      }
```

Comment:
This if block becomes dead code after adding the if blocking when types are identical